src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -234,15 +235,12 @@ func determineInternalIP() (string, error) {
 		return "", fmt.Errorf("Unable to determine local IP: %s", err)
 	}
 	defer conn.Close()
-	return strings.Split(conn.LocalAddr().String(), ":")[0], nil
+	host, _, err := net.SplitHostPort(conn.LocalAddr().String())
+	return host, err
 }
 
 func onBytesGiven(destAddr string, req *http.Request, bytes int64) {
-	port := "0"
-	parts := strings.Split(destAddr, ":")
-	if len(parts) > 1 {
-		port = parts[1]
-	}
+	_, port, _ := net.SplitHostPort(destAddr)
 
```

Comment:
So this is slightly different now in the case of a quirky address, as port will be the empty string instead of 0. Is that what you want?